sinks/opentsdb/driver.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentsdb
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+	"sync"
+
+	opentsdbclient "github.com/bluebreezecf/opentsdb-goclient/client"
+	opentsdbcfg "github.com/bluebreezecf/opentsdb-goclient/config"
+	"github.com/golang/glog"
+	"k8s.io/heapster/extpoints"
+	sink_api "k8s.io/heapster/sinks/api"
+	kube_api "k8s.io/kubernetes/pkg/api"
+)
+
+const (
+	defaultTagName   = "defaultTagName"
+	defaultTagValue  = "defaultTagValue"
+	eventMetricName  = "log_events"
+	eventUID         = "uid"
+	opentsdbSinkName = "OpenTSDB Sink"
+)
+
+var (
+	eventPodID   = sink_api.LabelPodId.Key
+	eventPodName = sink_api.LabelPodName.Key
+	eventHost    = sink_api.LabelHostname.Key
+)
+
+// openTSDBClient defines the minimal methods which will be used to
+// communicate with the target OpenTSDB for current openTSDBSink instance.
+type openTSDBClient interface {
+	Ping() error
+	Put(datapoints []opentsdbclient.DataPoint, queryParam string) (*opentsdbclient.PutResponse, error)
+}
+
+type openTSDBSink struct {
+	client openTSDBClient
+	sync.RWMutex
+	writeFailures int
+	config        opentsdbcfg.OpenTSDBConfig
+}
+
+func (tsdbSink *openTSDBSink) Register(metrics []sink_api.MetricDescriptor) error {
+	return nil
+}
+
+func (tsdbSink *openTSDBSink) Unregister(metrics []sink_api.MetricDescriptor) error {
+	return nil
+}
+
+func (tsdbSink *openTSDBSink) StoreTimeseries(timeseries []sink_api.Timeseries) error {
+	if timeseries == nil || len(timeseries) <= 0 {
+		return nil
+	}
+	if err := tsdbSink.client.Ping(); err != nil {
```

Comment:
Can you use the `[ClientInitializer](https://github.com/kubernetes/heapster/blob/master/sinks/util/client.go#L28)` interface to abstract out this `Ping`? 
Since this driver does not require any special setup, you can pass `func() {return nil}` as `initializer` and use this `tsdbSink.client.Ping` method as `ping` argument.
Example usage [here](https://github.com/kubernetes/heapster/blob/master/sinks/kafka/driver.go#L223).